flink/src/main/java/org/apache/iceberg/flink/sink/HybridKeySelector.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.iceberg.flink.sink;
+
+import java.util.List;
+import org.apache.flink.api.java.functions.KeySelector;
+import org.apache.flink.table.data.RowData;
+import org.apache.flink.table.types.logical.RowType;
+import org.apache.iceberg.PartitionSpec;
+import org.apache.iceberg.Schema;
+
+public class HybridKeySelector implements KeySelector<RowData, String> {
```

Comment:
Nit but also an important question to help me understand better =)

The name `HybridKeySelector` is admittedly somewhat confusing to me initially. Once I see the class definition, it's more understandable, but I do have to look through the source to see what `Hybrid` means.

Which leads me to my question:
In the case of current `HybridKeySelector`, we're using the PartitionKey and the `EqualityFieldKey`. This seems to work for `hash` distribution (at least per the updated tests), but what about `range` and `none`? Which key selectors would they use?

Reply:
I agree `HybridKeySelector` is not easy to understand. So I will rename it to `CombinedKeySelector`.

I think distribution mode and key selector is not one-to-one correspondence, different distribution mode will use 
`EqualityFieldKeySelector` or `PartitionKeySelector` or `CombinedKeySelector`. And `range` mode will not be consider here, cuz flink not support `range` mode yet.